PyISY/Nodes/node.py

```diff
@@ -168,9 +168,30 @@ def _get_notes(self):
                 self.parent.log.error('ISY Could not parse node ' + self._id + ' notes '
                                       + 'poorly formatted XML.')
             spoken_tag = notesdom.getElementsByTagName('spoken')
-            if spoken_tag:
+            if spoken_tag and len(spoken_tag) > 0 and spoken_tag[0].firstChild is not None:
                 self._spoken = spoken_tag[0].firstChild.toxml()
+            else:
+                self._spoken = None
         
+    def get_groups(self, controller=True, responder=True):
+        """
+        Returns the groups (scenes) that this node is a member of.
+        If controller is True, then the scene it controls is added to the list
+        If responder is True, then the scenes it is a responder of are added to the list
+        """
+        groups = []
+        for child in self.parent.parent.nodes.allLowerNodes:
```

Comment:
Instead of calling self.parent.parent.nodes, couldn't you call self.parent? I believe it should be the same thing.